### PR TITLE
Fix compareTensors for u4/i4 failure

### PR DESCRIPTION
**Pull request by WenjingKangIntel**
### Details:
 - Update the logic in `compareTensors` when bitwidth is less than 8.
 - Enable the OVInferRequestCheckTensorPrecision tests for i4/u4.

### Tickets:
 - *142251*

### CODE REVIEW

**Comment by XinWangIntel** [comment]
build_jenkins

**Comment by WenjingKangIntel** [comment]
build_jenkins

**Comment by WenjingKangIntel** [comment]
build_jenkins

**Comment by XinWangIntel** [comment]
build_jenkins

**Comment by XinWangIntel** [comment]
build_jenkins

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp [hunk 1]
```diff
@@ -380,9 +380,14 @@ bool OVInferRequestCheckTensorPrecision::compareTensors(const ov::Tensor& t1, co
...
-           t1.get_byte_size() == t2.get_byte_size() && t1.get_size() == t2.get_size() &&
-           t1.get_strides() == t2.get_strides() && data1 == data2;
+    if (t1.get_element_type().bitwidth() >= 8 && t2.get_element_type().bitwidth() >= 8) {
+        return t1.get_element_type() == t2.get_element_type() && t1.get_shape() == t2.get_shape() &&
+                t1.get_byte_size() == t2.get_byte_size() && t1.get_size() == t2.get_size() &&
+                t1.get_strides() == t2.get_strides() && data1 == data2;
+    } else {
+        return t1.get_element_type() == t2.get_element_type() && t1.get_shape() == t2.get_shape() &&
+               t1.get_byte_size() == t2.get_byte_size() && t1.get_size() == t2.get_size() && data1 == data2;
+    }
```
[the remark this answers is not in the document]
Comment (WenjingKangIntel) [reply src/tests/functional/plugin/shared/src/behavior/ov_infer_request/io_tensor.cpp]: Thanks for the comment. It's much neater now.

**Comment by WenjingKangIntel** [comment]
build_jenkins

**Review by praasz — APPROVED** [review]

**Review by xipingyan — APPROVED** [review]
LGTM

**Review by PatrikStepan — APPROVED** [review]

**Pull request merged**